javaagent-bootstrap/build.gradle.kts

```diff
@@ -25,6 +25,7 @@ dependencies {
 
   implementation(project(":instrumentation-api"))
   implementation(project(":javaagent-api"))
+  implementation(project(":instrumentation", configuration = "bootstrap"))
```

Comment:
I am very suspicious about this dependency... `javaagent-bootstrap` depending on `instrumentation` project is not something I would expect... Also, this will not work with extensions, right? I mean, instrumentation from extension will not be able to add bootstrap classes this way?

Reply:
> Also, this will not work with extensions, right? I mean, instrumentation from extension will not be able to add bootstrap classes this way?

No, it won't. Currently there's no way to add something to bootstrap with an extension, it has to be built into the javaagent jar.

> I am very suspicious about this dependency... `javaagent-bootstrap` depending on `instrumentation` project is not something I would expect...

Well, it depends on a specific configuration of `instrumentation` - but if you take that part away it might be confusing.

Reply:
For extensions we could package classes that are supposed to be loaded by agent class loader under some prefix like `inst/` in agent and use `Instrumentation.appendToBootstrapClassLoaderSearch` on extension jar if we find that extensions also need to add stuff too boot classpath.

Reply:
> For extensions we could package classes that are supposed to be loaded by agent class loader under some prefix like `inst/` in agent and use `Instrumentation.appendToBootstrapClassLoaderSearch` on extension jar if we find that extensions also need to add stuff too boot classpath.

That or create a new SPI (or extend `BootstrapPackagesConfigurer`) to enable marking whole packages as "should inject to bootstrap"